File: apps/mobile/ios/Sources/Components/TchatToast.swift
//
//  TchatToast.swift
//  TchatApp
//
//  Created by Claude on 22/09/2024.
//

import SwiftUI

/// Toast notification component following Tchat design system
public struct TchatToast: View {

    // MARK: - Toast Types
    public enum ToastType {
        case info
        case success
        case warning
        case error
    }

    public enum ToastPosition {
        case top
        case bottom
        case center
    }

    public enum ToastStyle {
        case filled
        case outlined
        case minimal
    }

    // MARK: - Properties
    @Binding private var isPresented: Bool
    @State private var dragOffset: CGSize = .zero
    @State private var animationOffset: CGFloat = 0

    let type: ToastType
    let position: ToastPosition
    let style: ToastStyle
    let message: String
    let icon: String?
    let duration: TimeInterval
    let isDismissible: Bool
    let hapticFeedback: Bool
    let onDismiss: (() -> Void)?

    // MARK: - Private Properties
    private let colors = Colors()
    private let dismissThreshold: CGFloat = 100

    // MARK: - Computed Properties
    private var toastColors: (background: Color, border: Color, text: Color, icon: Color) {
        switch (type, style) {
        case (.info, .filled):
            return (colors.primary, colors.primary, colors.textOnPrimary, colors.textOnPrimary)
        case (.info, .outlined):
            return (colors.background, colors.primary, colors.primary, colors.primary)
        case (.info, .minimal):
            return (colors.primary.opacity(0.1), Color.clear, colors.primary, colors.primary)

        case (.success, .filled):
            return (colors.success, colors.success, colors.textOnPrimary, colors.textOnPrimary)
        case (.success, .outlined):
            return (colors.background, colors.success, colors.success, colors.success)
        case (.success, .minimal):
            return (colors.success.opacity(0.1), Color.clear, colors.success, colors.success)

        case (.warning, .filled):
            return (colors.warning, colors.warning, colors.textOnPrimary, colors.textOnPrimary)
        case (.warning, .outlined):
            return (colors.background, colors.warning, colors.warning, colors.warning)
        case (.warning, .minimal):
            return (colors.warning.opacity(0.1), Color.clear, colors.warning, colors.warning)

        case (.error, .filled):
            return (colors.error, colors.error, colors.textOnPrimary, colors.textOnPrimary)
        case (.error, .outlined):
            return (colors.background, colors.error, colors.error, colors.error)
        case (.error, .minimal):
            return (colors.error.opacity(0.1), Color.clear, colors.error, colors.error)
        }
    }

    private var defaultIcon: String {
        switch type {
        case .info: return "info.circle.fill"
        case .success: return "checkmark.circle.fill"
        case .warning: return "exclamationmark.triangle.fill"
        case .error: return "xmark.circle.fill"
        }
    }

    private var displayIcon: String {
        icon ?? defaultIcon
    }

    private var initialOffset: CGFloat {
        switch position {
        case .top: return -200
        case .bottom: return 200
        case .center: return 0
        }
    }

    // MARK: - Initializer
    public init(
        isPresented: Binding<Bool>,
        type: ToastType,
        position: ToastPosition = .top,
        style: ToastStyle = .filled,
        message: String,
        icon: String? = nil,
        duration: TimeInterval = 3.0,
        isDismissible: Bool = true,
        hapticFeedback: Bool = true,
        onDismiss: (() -> Void)? = nil
    ) {
        self._isPresented = isPresented
        self.type = type
        self.position = position
        self.style = style
        self.message = message
        self.icon = icon
        self.duration = duration
        self.isDismissible = isDismissible
        self.hapticFeedback = hapticFeedback
        self.onDismiss = onDismiss
    }

    // MARK: - Body
    public var body: some View {
        if isPresented {
            toastContent
                .transition(.asymmetric(
                    insertion: .move(edge: position == .top ? .top : .bottom).combined(with: .opacity),
                    removal: .move(edge: position == .top ? .top : .bottom).combined(with: .opacity)
                ))
                .onAppear {
                    if hapticFeedback {
                        triggerHapticFeedback()
                    }

                    withAnimation(.easeOut(duration: 0.4)) {
                        animationOffset = 0
                    }

                    // Auto-dismiss after duration
                    if duration > 0 {
                        DispatchQueue.main.asyncAfter(deadline: .now() + duration) {
                            dismissToast()
                        }
                    }
                }
        }
    }

    // MARK: - Toast Content
    @ViewBuilder
    private var toastContent: some View {
        HStack(spacing: Spacing.sm) {
            // Icon
            Image(systemName: displayIcon)
                .font(.system(size: 16, weight: .semibold))
                .foregroundColor(toastColors.icon)

            // Message
            Text(message)
                .font(.system(size: 14, weight: .medium))
                .foregroundColor(toastColors.text)
                .multilineTextAlignment(.leading)
                .frame(maxWidth: .infinity, alignment: .leading)

            // Dismiss button (optional)
            if isDismissible {
                Button(action: dismissToast) {
                    Image(systemName: "xmark")
                        .font(.system(size: 12, weight: .medium))
                        .foregroundColor(toastColors.text.opacity(0.7))
                }
            }
        }
        .padding(.horizontal, Spacing.md)
        .padding(.vertical, Spacing.sm)
        .background(toastColors.background)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(toastColors.border, lineWidth: style == .outlined ? 1 : 0)
        )
        .cornerRadius(8)
        .shadow(color: colors.shadowMedium, radius: 8, y: 4)
        .padding(.horizontal, Spacing.md)
        .offset(x: dragOffset.width, y: animationOffset + dragOffset.height)
        .gesture(
            isDismissible ? dragGesture : nil
        )
    }

    // MARK: - Drag Gesture
    private var dragGesture: some Gesture {
        DragGesture()
            .onChanged { value in
                dragOffset = value.translation
            }
            .onEnded { value in
                let shouldDismiss = abs(value.translation.width) > dismissThreshold ||
                                  abs(value.translation.height) > dismissThreshold

                if shouldDismiss {
                    withAnimation(.easeIn(duration: 0.2)) {
                        dragOffset = CGSize(
                            width: value.translation.width > 0 ? 400 : -400,
                            height: value.translation.height
                        )
                    }

                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
                        dismissToast()
                    }
                } else {
                    withAnimation(.spring()) {
                        dragOffset = .zero
                    }
                }
            }
    }

    // MARK: - Actions
    private func dismissToast() {
        withAnimation(.easeIn(duration: 0.3)) {
            animationOffset = initialOffset
        }

        DispatchQueue.main.asyncAfter(deadline: .now() + 0.3) {
            isPresented = false
            onDismiss?()
        }
    }

    private func triggerHapticFeedback() {
        let feedbackStyle: UIImpactFeedbackGenerator.FeedbackStyle

        switch type {
        case .success:
            feedbackStyle = .light
        case .warning:
            feedbackStyle = .medium
        case .error:
            feedbackStyle = .heavy
        case .info:
            feedbackStyle = .light
        }

        let impactFeedback = UIImpactFeedbackGenerator(style: feedbackStyle)
        impactFeedback.impactOccurred()
    }
}

// MARK: - Toast Manager
@MainActor
public class TchatToastManager: ObservableObject {
    @Published public var currentToasts: [TchatToastItem] = []

    public static let shared = TchatToastManager()

    private init() {}

    public func show(
        type: TchatToast.ToastType,
        position: TchatToast.ToastPosition = .top,
        style: TchatToast.ToastStyle = .filled,
        message: String,
        icon: String? = nil,
        duration: TimeInterval = 3.0,
        isDismissible: Bool = true,
        hapticFeedback: Bool = true
    ) {
        let toast = TchatToastItem(
            type: type,
            position: position,
            style: style,
            message: message,
            icon: icon,
            duration: duration,
            isDismissible: isDismissible,
            hapticFeedback: hapticFeedback
        )

        currentToasts.append(toast)

        // Auto-dismiss after duration
        if duration > 0 {
            DispatchQueue.main.asyncAfter(deadline: .now() + duration + 0.5) {
                self.dismiss(toast.id)
            }
        }
    }

    public func dismiss(_ id: UUID) {
        currentToasts.removeAll { $0.id == id }
    }

    public func dismissAll() {
        currentToasts.removeAll()
    }
}

// MARK: - Toast Item
public struct TchatToastItem: Identifiable {
    public let id = UUID()
    let type: TchatToast.ToastType
    let position: TchatToast.ToastPosition
    let style: TchatToast.ToastStyle
    let message: String
    let icon: String?
    let duration: TimeInterval
    let isDismissible: Bool
    let hapticFeedback: Bool

    public init(
        type: TchatToast.ToastType,
        position: TchatToast.ToastPosition = .top,
        style: TchatToast.ToastStyle = .filled,
        message: String,
        icon: String? = nil,
        duration: TimeInterval = 3.0,
        isDismissible: Bool = true,
        hapticFeedback: Bool = true
    ) {
        self.type = type
        self.position = position
        self.style = style
        self.message = message
        self.icon = icon
        self.duration = duration
        self.isDismissible = isDismissible
        self.hapticFeedback = hapticFeedback
    }
}

// MARK: - Toast Overlay
public struct TchatToastOverlay: View {
    @ObservedObject private var toastManager = TchatToastManager.shared

    public init() {}

    public var body: some View {
        GeometryReader { geometry in
            ZStack {
                // Top toasts
                VStack(spacing: Spacing.xs) {
                    ForEach(toastManager.currentToasts.filter { $0.position == .top }) { toast in
                        TchatToast(
                            isPresented: .constant(true),
                            type: toast.type,
                            position: toast.position,
                            style: toast.style,
                            message: toast.message,
                            icon: toast.icon,
                            duration: 0, // Managed by toast manager
                            isDismissible: toast.isDismissible,
                            hapticFeedback: false // Already triggered
                        ) {
                            toastManager.dismiss(toast.id)
                        }
                    }

                    Spacer()
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .top)
                .padding(.top, geometry.safeAreaInsets.top + Spacing.md)

                // Center toasts
                VStack(spacing: Spacing.xs) {
                    ForEach(toastManager.currentToasts.filter { $0.position == .center }) { toast in
                        TchatToast(
                            isPresented: .constant(true),
                            type: toast.type,
                            position: toast.position,
                            style: toast.style,
                            message: toast.message,
                            icon: toast.icon,
                            duration: 0,
                            isDismissible: toast.isDismissible,
                            hapticFeedback: false
                        ) {
                            toastManager.dismiss(toast.id)
                        }
                    }
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .center)

                // Bottom toasts
                VStack(spacing: Spacing.xs) {
                    Spacer()

                    ForEach(toastManager.currentToasts.filter { $0.position == .bottom }.reversed()) { toast in
                        TchatToast(
                            isPresented: .constant(true),
                            type: toast.type,
                            position: toast.position,
                            style: toast.style,
                            message: toast.message,
                            icon: toast.icon,
                            duration: 0,
                            isDismissible: toast.isDismissible,
                            hapticFeedback: false
                        ) {
                            toastManager.dismiss(toast.id)
                        }
                    }
                }
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .bottom)
                .padding(.bottom, geometry.safeAreaInsets.bottom + Spacing.md)
            }
        }
        .allowsHitTesting(false)
        .animation(.easeInOut(duration: 0.3), value: toastManager.currentToasts.count)
    }
}

// MARK: - Convenience Methods
extension TchatToastManager {
    public func showSuccess(
        _ message: String,
        position: TchatToast.ToastPosition = .top,
        duration: TimeInterval = 2.0
    ) {
        show(
            type: .success,
            position: position,
            message: message,
            duration: duration
        )
    }

    public func showError(
        _ message: String,
        position: TchatToast.ToastPosition = .top,
        duration: TimeInterval = 4.0
    ) {
        show(
            type: .error,
            position: position,
            message: message,
            duration: duration
        )
    }

    public func showWarning(
        _ message: String,
        position: TchatToast.ToastPosition = .top,
        duration: TimeInterval = 3.0
    ) {
        show(
            type: .warning,
            position: position,
            message: message,
            duration: duration
        )
    }

    public func showInfo(
        _ message: String,
        position: TchatToast.ToastPosition = .top,
        duration: TimeInterval = 3.0
    ) {
        show(
            type: .info,
            position: position,
            message: message,
            duration: duration
        )
    }
}

// MARK: - Preview
#if DEBUG
struct TchatToast_Previews: PreviewProvider {
    static var previews: some View {
        ZStack {
            Color(.systemGray6)
                .ignoresSafeArea()

            VStack(spacing: Spacing.lg) {
                // Success toast
                TchatToast(
                    isPresented: .constant(true),
                    type: .success,
                    position: .top,
                    style: .filled,
                    message: "Changes saved successfully!",
                    duration: 0
                )

                // Error toast
                TchatToast(
                    isPresented: .constant(true),
                    type: .error,
                    position: .top,
                    style: .outlined,
                    message: "Failed to upload file. Please try again.",
                    duration: 0
                )

                // Warning toast
                TchatToast(
                    isPresented: .constant(true),
                    type: .warning,
                    position: .top,
                    style: .minimal,
                    message: "You have unsaved changes.",
                    duration: 0
                )

                // Info toast with custom icon
                TchatToast(
                    isPresented: .constant(true),
                    type: .info,
                    position: .top,
                    style: .filled,
                    message: "New update available for download.",
                    icon: "arrow.down.circle.fill",
                    duration: 0,
                    isDismissible: false
                )
            }
            .padding()
        }
        .previewLayout(.sizeThatFits)
    }
}
#endif

File: apps/mobile/android/app/src/main/java/com/tchat/components/TchatToast.kt
package com.tchat.components

import androidx.compose.animation.*
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.gestures.detectDragGestures
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.input.pointer.pointerInput
import androidx.compose.ui.platform.LocalDensity
import androidx.compose.ui.platform.LocalHapticFeedback
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.IntOffset
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tchat.designsystem.Colors
import com.tchat.designsystem.Spacing
import kotlinx.coroutines.delay
import kotlin.math.roundToInt

/**
 * Toast notification component following Tchat design system
 */
@Composable
fun TchatToast(
    isVisible: Boolean,
    type: TchatToastType,
    message: String,
    modifier: Modifier = Modifier,
    position: TchatToastPosition = TchatToastPosition.Top,
    style: TchatToastStyle = TchatToastStyle.Filled,
    icon: ImageVector? = null,
    duration: Long = 3000L,
    isDismissible: Boolean = true,
    hapticFeedback: Boolean = true,
    onDismiss: (() -> Unit)? = null
) {
    val haptic = LocalHapticFeedback.current
    val density = LocalDensity.current

    var dragOffset by remember { mutableStateOf(0f) }
    val dismissThreshold = with(density) { 100.dp.toPx() }

    val toastColors = getToastColors(type, style)
    val toastIcon = icon ?: getDefaultIcon(type)

    LaunchedEffect(isVisible) {
        if (isVisible) {
            if (hapticFeedback) {
                val feedbackType = when (type) {
                    TchatToastType.Success -> androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                    TchatToastType.Error -> androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                    TchatToastType.Warning -> androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                    TchatToastType.Info -> androidx.compose.ui.hapticfeedback.HapticFeedbackType.LongPress
                }
                haptic.performHapticFeedback(feedbackType)
            }

            if (duration > 0) {
                delay(duration)
                onDismiss?.invoke()
            }
        }
    }

    AnimatedVisibility(
        visible = isVisible,
        enter = slideInVertically(
            initialOffsetY = { if (position == TchatToastPosition.Top) -it else it },
            animationSpec = tween(400)
        ) + fadeIn(animationSpec = tween(400)),
        exit = slideOutVertically(
            targetOffsetY = { if (position == TchatToastPosition.Top) -it else it },
            animationSpec = tween(300)
        ) + fadeOut(animationSpec = tween(300))
    ) {
        Row(
            modifier = modifier
                .fillMaxWidth()
                .offset { IntOffset(dragOffset.roundToInt(), 0) }
                .shadow(
                    elevation = 8.dp,
                    shape = RoundedCornerShape(8.dp)
                )
                .background(
                    color = toastColors.background,
                    shape = RoundedCornerShape(8.dp)
                )
                .then(
                    if (style == TchatToastStyle.Outlined) {
                        Modifier.border(
                            width = 1.dp,
                            color = toastColors.border,
                            shape = RoundedCornerShape(8.dp)
                        )
                    } else Modifier
                )
                .padding(horizontal = Spacing.md, vertical = Spacing.sm)
                .then(
                    if (isDismissible) {
                        Modifier.pointerInput(Unit) {
                            detectDragGestures(
                                onDragEnd = {
                                    val shouldDismiss = kotlin.math.abs(dragOffset) > dismissThreshold
                                    if (shouldDismiss) {
                                        onDismiss?.invoke()
                                    } else {
                                        dragOffset = 0f
                                    }
                                }
                            ) { _, dragAmount ->
                                dragOffset += dragAmount.x
                            }
                        }
                    } else Modifier
                ),
            horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Icon
            Icon(
                imageVector = toastIcon,
                contentDescription = null,
                tint = toastColors.icon,
                modifier = Modifier.size(16.dp)
            )

            // Message
            Text(
                text = message,
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                color = toastColors.text,
                modifier = Modifier.weight(1f)
            )

            // Dismiss button
            if (isDismissible) {
                Icon(
                    imageVector = Icons.Default.Close,
                    contentDescription = "Dismiss",
                    tint = toastColors.text.copy(alpha = 0.7f),
                    modifier = Modifier
                        .size(16.dp)
                        .clickable { onDismiss?.invoke() }
                )
            }
        }
    }
}

private fun getToastColors(type: TchatToastType, style: TchatToastStyle): ToastColors {
    return when (type to style) {
        TchatToastType.Info to TchatToastStyle.Filled -> ToastColors(
            background = Colors.primary,
            border = Colors.primary,
            text = Colors.textOnPrimary,
            icon = Colors.textOnPrimary
        )
        TchatToastType.Info to TchatToastStyle.Outlined -> ToastColors(
            background = Colors.background,
            border = Colors.primary,
            text = Colors.primary,
            icon = Colors.primary
        )
        TchatToastType.Info to TchatToastStyle.Minimal -> ToastColors(
            background = Colors.primary.copy(alpha = 0.1f),
            border = Color.Transparent,
            text = Colors.primary,
            icon = Colors.primary
        )

        TchatToastType.Success to TchatToastStyle.Filled -> ToastColors(
            background = Colors.success,
            border = Colors.success,
            text = Colors.textOnPrimary,
            icon = Colors.textOnPrimary
        )
        TchatToastType.Success to TchatToastStyle.Outlined -> ToastColors(
            background = Colors.background,
            border = Colors.success,
            text = Colors.success,
            icon = Colors.success
        )
        TchatToastType.Success to TchatToastStyle.Minimal -> ToastColors(
            background = Colors.success.copy(alpha = 0.1f),
            border = Color.Transparent,
            text = Colors.success,
            icon = Colors.success
        )

        TchatToastType.Warning to TchatToastStyle.Filled -> ToastColors(
            background = Colors.warning,
            border = Colors.warning,
            text = Colors.textOnPrimary,
            icon = Colors.textOnPrimary
        )
        TchatToastType.Warning to TchatToastStyle.Outlined -> ToastColors(
            background = Colors.background,
            border = Colors.warning,
            text = Colors.warning,
            icon = Colors.warning
        )
        TchatToastType.Warning to TchatToastStyle.Minimal -> ToastColors(
            background = Colors.warning.copy(alpha = 0.1f),
            border = Color.Transparent,
            text = Colors.warning,
            icon = Colors.warning
        )

        TchatToastType.Error to TchatToastStyle.Filled -> ToastColors(
            background = Colors.error,
            border = Colors.error,
            text = Colors.textOnPrimary,
            icon = Colors.textOnPrimary
        )
        TchatToastType.Error to TchatToastStyle.Outlined -> ToastColors(
            background = Colors.background,
            border = Colors.error,
            text = Colors.error,
            icon = Colors.error
        )
        TchatToastType.Error to TchatToastStyle.Minimal -> ToastColors(
            background = Colors.error.copy(alpha = 0.1f),
            border = Color.Transparent,
            text = Colors.error,
            icon = Colors.error
        )
        else -> ToastColors(
            background = Colors.background,
            border = Colors.border,
            text = Colors.textPrimary,
            icon = Colors.textPrimary
        )
    }
}

private fun getDefaultIcon(type: TchatToastType): ImageVector {
    return when (type) {
        TchatToastType.Info -> Icons.Default.Info
        TchatToastType.Success -> Icons.Default.CheckCircle
        TchatToastType.Warning -> Icons.Default.Warning
        TchatToastType.Error -> Icons.Default.Error
    }
}

/**
 * Toast manager for global toast handling
 */
object TchatToastManager {
    private var _currentToasts = mutableStateOf<List<TchatToastItem>>(emptyList())
    val currentToasts: State<List<TchatToastItem>> = _currentToasts

    fun show(
        type: TchatToastType,
        message: String,
        position: TchatToastPosition = TchatToastPosition.Top,
        style: TchatToastStyle = TchatToastStyle.Filled,
        icon: ImageVector? = null,
        duration: Long = 3000L,
        isDismissible: Boolean = true,
        hapticFeedback: Boolean = true
    ) {
        val toast = TchatToastItem(
            type = type,
            message = message,
            position = position,
            style = style,
            icon = icon,
            duration = duration,
            isDismissible = isDismissible,
            hapticFeedback = hapticFeedback
        )

        _currentToasts.value = _currentToasts.value + toast
    }

    fun dismiss(id: String) {
        _currentToasts.value = _currentToasts.value.filter { it.id != id }
    }

    fun dismissAll() {
        _currentToasts.value = emptyList()
    }

    fun showSuccess(
        message: String,
        position: TchatToastPosition = TchatToastPosition.Top,
        duration: Long = 2000L
    ) {
        show(
            type = TchatToastType.Success,
            message = message,
            position = position,
            duration = duration
        )
    }

    fun showError(
        message: String,
        position: TchatToastPosition = TchatToastPosition.Top,
        duration: Long = 4000L
    ) {
        show(
            type = TchatToastType.Error,
            message = message,
            position = position,
            duration = duration
        )
    }

    fun showWarning(
        message: String,
        position: TchatToastPosition = TchatToastPosition.Top,
        duration: Long = 3000L
    ) {
        show(
            type = TchatToastType.Warning,
            message = message,
            position = position,
            duration = duration
        )
    }

    fun showInfo(
        message: String,
        position: TchatToastPosition = TchatToastPosition.Top,
        duration: Long = 3000L
    ) {
        show(
            type = TchatToastType.Info,
            message = message,
            position = position,
            duration = duration
        )
    }
}

/**
 * Toast overlay composable for global toasts
 */
@Composable
fun TchatToastOverlay() {
    val currentToasts by TchatToastManager.currentToasts

    BoxWithConstraints(
        modifier = Modifier.fillMaxSize()
    ) {
        // Top toasts
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .align(Alignment.TopCenter)
                .padding(horizontal = Spacing.md)
                .padding(top = Spacing.md),
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            currentToasts
                .filter { it.position == TchatToastPosition.Top }
                .forEach { toast ->
                    key(toast.id) {
                        TchatToast(
                            isVisible = true,
                            type = toast.type,
                            message = toast.message,
                            position = toast.position,
                            style = toast.style,
                            icon = toast.icon,
                            duration = 0, // Managed by toast manager
                            isDismissible = toast.isDismissible,
                            hapticFeedback = false, // Already triggered
                            onDismiss = { TchatToastManager.dismiss(toast.id) }
                        )
                    }
                }
        }

        // Center toasts
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .align(Alignment.Center)
                .padding(horizontal = Spacing.md),
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            currentToasts
                .filter { it.position == TchatToastPosition.Center }
                .forEach { toast ->
                    key(toast.id) {
                        TchatToast(
                            isVisible = true,
                            type = toast.type,
                            message = toast.message,
                            position = toast.position,
                            style = toast.style,
                            icon = toast.icon,
                            duration = 0,
                            isDismissible = toast.isDismissible,
                            hapticFeedback = false,
                            onDismiss = { TchatToastManager.dismiss(toast.id) }
                        )
                    }
                }
        }

        // Bottom toasts
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .align(Alignment.BottomCenter)
                .padding(horizontal = Spacing.md)
                .padding(bottom = Spacing.md),
            verticalArrangement = Arrangement.spacedBy(Spacing.xs)
        ) {
            currentToasts
                .filter { it.position == TchatToastPosition.Bottom }
                .reversed() // Show newest at bottom
                .forEach { toast ->
                    key(toast.id) {
                        TchatToast(
                            isVisible = true,
                            type = toast.type,
                            message = toast.message,
                            position = toast.position,
                            style = toast.style,
                            icon = toast.icon,
                            duration = 0,
                            isDismissible = toast.isDismissible,
                            hapticFeedback = false,
                            onDismiss = { TchatToastManager.dismiss(toast.id) }
                        )
                    }
                }
        }
    }
}

/**
 * Data classes and enums
 */
data class TchatToastItem(
    val id: String = java.util.UUID.randomUUID().toString(),
    val type: TchatToastType,
    val message: String,
    val position: TchatToastPosition,
    val style: TchatToastStyle,
    val icon: ImageVector?,
    val duration: Long,
    val isDismissible: Boolean,
    val hapticFeedback: Boolean
)

enum class TchatToastType {
    Info,
    Success,
    Warning,
    Error
}

enum class TchatToastPosition {
    Top,
    Center,
    Bottom
}

enum class TchatToastStyle {
    Filled,
    Outlined,
    Minimal
}

private data class ToastColors(
    val background: Color,
    val border: Color,
    val text: Color,
    val icon: Color
)

// Preview
@Preview(showBackground = true)
@Composable
fun TchatToastPreview() {
    Column(
        modifier = Modifier.padding(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.lg)
    ) {
        // Success toast
        TchatToast(
            isVisible = true,
            type = TchatToastType.Success,
            style = TchatToastStyle.Filled,
            message = "Changes saved successfully!",
            duration = 0
        )

        // Error toast
        TchatToast(
            isVisible = true,
            type = TchatToastType.Error,
            style = TchatToastStyle.Outlined,
            message = "Failed to upload file. Please try again.",
            duration = 0
        )

        // Warning toast
        TchatToast(
            isVisible = true,
            type = TchatToastType.Warning,
            style = TchatToastStyle.Minimal,
            message = "You have unsaved changes.",
            duration = 0
        )

        // Info toast with custom icon
        TchatToast(
            isVisible = true,
            type = TchatToastType.Info,
            style = TchatToastStyle.Filled,
            message = "New update available for download.",
            icon = Icons.Default.Download,
            duration = 0,
            isDismissible = false
        )
    }
}